File: 2018-10-20-fizz-buzz-go/src/fizzbuzz/fizzbuzz_test.go
package fizzbuzz

import "testing"

func TestFizzBuzz(t *testing.T) {
	cases := []struct {
		in   int
		want string
	}{
		{1, "1"},
		{3, "fizz"},
		{5, "buzz"},
		{3 * 5, "fizzbuzz"},
		{3 * 5 * 2, "fizzbuzz"},
		{7, "bizz"},
		{3 * 7, "fizzbizz"},
		{3 * 7 * 2, "fizzbizz"},
		{5 * 7, "buzzbizz"},
		{5 * 7 * 2, "buzzbizz"},
		{3 * 5 * 7, "fizzbuzzbizz"},
		{3 * 5 * 7 * 2, "fizzbuzzbizz"},
	}
	for _, c := range cases {
		got := FizzBuzz(c.in)
		if got != c.want {
			t.Errorf("FizzBuzz(%q) == %q, want %q", c.in, got, c.want)
		}
	}
}

File: 2018/2018-10-19-fiz-buzz-python/fizz_buzz_test.py
import unittest
from fizz_buzz import fizzBuzz

class TestFizzBuzz(unittest.TestCase):

    def test_other(self):
        self.assertEqual(fizzBuzz(1), 1)
        self.assertEqual(fizzBuzz(2), 2)
    
    def test_3(self):
        self.assertEqual(fizzBuzz(3), 'fizz')
        self.assertEqual(fizzBuzz(6), 'fizz')
        self.assertEqual(fizzBuzz(9), 'fizz')
    
    def test_5(self):
        self.assertEqual(fizzBuzz(5), 'buzz')
        self.assertEqual(fizzBuzz(10), 'buzz')

    def test_7(self):
        self.assertEqual(fizzBuzz(7), 'bizz')
        self.assertEqual(fizzBuzz(7 * 2), 'bizz')
    
    def test_3_7(self):
        self.assertEqual(fizzBuzz(3 * 7), 'fizzbizz')
        self.assertEqual(fizzBuzz(3 * 7 * 2), 'fizzbizz')

    def test_5_7(self):
        self.assertEqual(fizzBuzz(5 * 7), 'buzzbizz')
        self.assertEqual(fizzBuzz(5 * 7 * 2), 'buzzbizz')
    
    def test_3_5_7(self):
        self.assertEqual(fizzBuzz(3 * 5 * 7), 'fizzbuzzbizz')
        self.assertEqual(fizzBuzz(3 * 5 * 7 * 2), 'fizzbuzzbizz')
    
    def test_everything(self):
        self.assertEqual(fizzBuzz(3 * 5 * 7 * 11 * 13 * 17 * 19 * 23 * 29 * 31 * 37), 'fizzbuzzbizzbazzboombingbangbongfozzfazzwoof')
        self.assertEqual(fizzBuzz(3 * 5 * 7 * 11 * 13 * 17 * 19 * 23 * 29 * 31 * 37 * 2), 'fizzbuzzbizzbazzboombingbangbongfozzfazzwoof')